sporkfish/searcher/searcher_factory.py

```diff
@@ -0,0 +1,33 @@
+from typing import Type
+
+from sporkfish.evaluator import Evaluator
+
+from .move_ordering import MoveOrder, MvvLvaHeuristic
+from .negamax import NegaMaxLazySmp, NegamaxSp
+from .searcher import Searcher
+from .searcher_config import MoveOrdering, SearcherConfig, SearchMode
+
+
+class SearcherFactory:
+    """
+    Factory class for creating instances of different searcher types.
+    """
+
+    # ideally would build evaluator as well
+    @staticmethod
+    def create(searcher_cfg: SearcherConfig, evaluator: Evaluator) -> Searcher:
```

Comment:
Discussed offline, perhaps in future (or this PR?) we will instantiate evaluator inside create. This is so we properly separate the concern of creating the evaluator from the client. In addition, we may not need an evaluator in the first place (e.g. MCTS), so the evaluator only needs to be created for the searchers that need it.

Reply:
Happy to do it in this PR

Reply:
Thanks!!

Reply:
Done - of course without evaluator switching, my change is a bit surface level. Have added comments to remind whoever going to work on evaluator switching #88 to be aware.

Reply:
Lovely :)